### ISSUE TITLE
AWS Lambda support for Java 11, Node.js 12 and Python 3.8

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#10919]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

> You can now develop AWS Lambda functions using Java 11.
> You can now author your AWS Lambda functions in Node.js 12.
> You can now develop your AWS Lambda functions using Python 3.8.

### New or Affected Resource(s)

* [`aws_lambda_function`](https://www.terraform.io/docs/providers/aws/r/lambda_function.html)
* [`aws_lambda_layer_version`](https://www.terraform.io/docs/providers/aws/r/lambda_layer_version.html) plus corresponding data source

### Potential Terraform Configuration

```hcl
resource "aws_lambda_function" "example" {
  runtime = "python3.8"
}
```

### References

Announcements:
* [Java 11](https://aws.amazon.com/about-aws/whats-new/2019/11/aws-lambda-supports-java-11/)
* [Node.js 12](https://aws.amazon.com/about-aws/whats-new/2019/11/aws-lambda-supports-node-js-12/)
* [Python 3.8](https://aws.amazon.com/about-aws/whats-new/2019/11/aws-lambda-now-supports-python-3-8/)

**Comment by DarkMukke** [comment]
for nodejs 12, this is a direct upgrade from #8658

**Comment by DarkMukke** [comment]
Is the fix for NodeJS not just very similar to https://github.com/terraform-providers/terraform-provider-aws/pull/8622/commits/6d39f615e685fc79216d53421e94d592c872cf1b ?

**Comment by cakejelly** [comment]
@ewbankkit If I understand correctly then this would mean that we just need to drop the runtime validations completely from this library and it should work? Or am I missing something

**Comment by DrFaust92** [comment]
@ewbankkit, it seems that [v1.25.38](https://github.com/aws/aws-sdk-go/releases/tag/v1.25.38) adds this

**Comment by ewbankkit** [comment]
@DrFaust92 Indeed it does; Thanks.
Requires:
* https://github.com/terraform-providers/terraform-provider-aws/pull/10940

**Comment by mischuler** [comment]
@ewbankkit 
Can we also add `aws_lambda_layer_version` as an affected resource? It also requires a runtime input.

```
resource "aws_lambda_layer_version" "imagemagick_layer" {
  filename   = "../layers-files/vdem-imagemagick-layer.zip"
  layer_name = "node12-imagemagick-layer"

  source_code_hash = "${filebase64sha256("../layers-files/node12-imagemagick-layer.zip")}"

  compatible_runtimes = ["nodejs12 .x"]
}
```


`Error: aws_lambda_layer_version.imagemagick_layer: expected compatible_runtimes.0 to be one of [dotnetcore1.0 dotnetcore2.0 dotnetcore2.1 go1.x java8 nodejs4.3 nodejs4.3-edge nodejs6.10 nodejs8.10 nodejs10.x provided python2.7 python3.6 python3.7 ruby2.5], got nodejs12.x`

**Comment by ewbankkit** [comment]
@mischuler Added `aws_lambda_layer_version` as affected resource (and data source); Thanks.

**Comment by bflad** [comment]
Support for `aws_lambda_function` has been merged and will release in version 2.39.0 of the Terraform AWS Provider, likely tomorrow. Will reopen to allow for `aws_lambda_layer_version` implementation.

**Comment by bflad** [comment]
Hi folks 👋 The runtime validation is actually shared between both `aws_lambda_function` and `aws_lambda_layer_version` resources and we have verified the updated validation in the latter resource as well. The CHANGELOG is updated to reflect both these resources receiving this update and this will release later today in version 2.39.0. 👍
